app/controllers/public_health_controller.rb

```diff
@@ -493,161 +442,61 @@ def advanced_filter(patients, filters)
+      return patients if timespan.nil?
+
+      timeframe = { after: timespan.ago.beginning_of_day, before: DateTime.now } if filter[:value][:when] == 'past'
+      timeframe = { after: DateTime.now, before: timespan.from_now.end_of_day } if filter[:value][:when] == 'next'
     end
-    patients
+    return patients if timeframe.nil?
+
+    patients_by_field_timeframe(patients, field, timeframe, tz_offset, type)
   end
 
-  def advanced_filter_lde_relative(patients, filter)
-    case filter[:relativeOption]
-    when 'today'
-      patients = patients.where('Date(last_date_of_exposure) = ?', Date.today)
-    when 'tomorrow'
-      patients = patients.where('Date(last_date_of_exposure) = ?', Date.tomorrow)
-    when 'yesterday'
-      patients = patients.where('Date(last_date_of_exposure) = ?', Date.yesterday)
-    when 'custom'
-      case filter[:value][:when]
-      when 'past'
-        case filter[:value][:unit]
-        when 'days'
-          number = filter[:value][:number].to_i
-          patients = patients.where('Date(last_date_of_exposure) >= ?', number.days.ago).where('Date(last_date_of_exposure) <= ?', Date.today)
-        when 'weeks'
-          number = filter[:value][:number].to_i
-          patients = patients.where('Date(last_date_of_exposure) >= ?', number.weeks.ago).where('Date(last_date_of_exposure) <= ?', Date.today)
-        when 'months'
-          number = filter[:value][:number].to_i
-          patients = patients.where('Date(last_date_of_exposure) >= ?', number.months.ago).where('Date(last_date_of_exposure) <= ?', Date.today)
-        end
-      when 'next'
-        case filter[:value][:unit]
-        when 'days'
-          number = filter[:value][:number].to_i
-          patients = patients.where('Date(last_date_of_exposure) <= ?', number.days.from_now).where('Date(last_date_of_exposure) >= ?', Date.today)
-        when 'weeks'
-          number = filter[:value][:number].to_i
-          patients = patients.where('Date(last_date_of_exposure) <= ?', number.weeks.from_now).where('Date(last_date_of_exposure) >= ?', Date.today)
-        when 'months'
-          number = filter[:value][:number].to_i
-          patients = patients.where('Date(last_date_of_exposure) <= ?', number.months.from_now).where('Date(last_date_of_exposure) >= ?', Date.today)
-        end
-      end
+  def patients_by_field_timeframe(patients, field, timeframe, tz_offset, type)
+    # Adjust for difference between client and server timezones.
+    # NOTE: Adding server timezone offset in cases where the server may not be running in UTC time.
+    # NOTE: + because js and ruby offsets are flipped. Both of these values are in seconds.
+    tz_diff = tz_offset.to_i.minutes + DateTime.now.utc_offset
+
+    if timeframe[:after].present?
+      # Convert timeframe value to date if field is a date, apply timezone difference if field is a datetime
+      after = type == :date ? timeframe[:after].to_date : timeframe[:after] - tz_diff
```

Comment:
Unapproving until we resolve this issue. This is actually still problematic because we unfortunately cannot ignore timezone in the case where the stored value is just a date with no timestamp. 

Here’s a way to replicate the issue in the console using the code that’s there now:

1. Assume normal date option for LDE was chosen and they chose the “after” comparison for today 12/8. At the time of writing this, it’s 12/8 still in EST, but it’s already the next day (12/9) in UTC time.
2. At this point, `timeframe[:after]` will be equal to: 
```
Chronic.parse('today').beginning_of_day + 1.day
=> 2020-12-09 00:00:00 -0500
```
(This output is confusing because that’s a UTC timestamp with a EST offset - I don't know why it does that when run locally - but on prod that will actually be a UTC offset).

3.  Because we’re checking the type here before applying timezone subtraction, `tz_diff` will be ignored and it will compared last_date_of_exposure to 12/9 rather than 12/8

We know that the date that is passed to the query for comparison will be converted to UTC no matter what by ActiveRecord. Essentially, we need that value that is determined by the user to never get converted to UTC for comparison to things like LDE. Therefore, it needs to be passed in such a way that after that automatic conversion is done, it is the date that the user entered still (so 12/8 and not 12/9 in this example).

Reply:
talked to @kylietmo about this already, but just for record keeping, here is what we came up with for a fix: https://github.com/SaraAlert/SaraAlert/pull/557/commits/0c8c7f633975576b2a4e331d337c5c979712cffd

Reply:
Testing, and this appears to be fixed. Resolving.